include/aspect/material_model/utilities.h

```diff
@@ -268,6 +268,29 @@ namespace aspect
                             const std::vector<double> &parameter_values,
                             const CompositionalAveragingOperation &average_type);
 
+      /**
+      * For multicomponent material models:
+      * Material models compute output quantities such as the viscosity, the
+      * density, etc. For some models, these values may depend on the phase in
+      * addition to the composition, and more than one phase field might have
+      * nonzero values at a given quadrature point. This means that properties
+      * for each composition have to be averaged based on the fractions of each
+      * phase field present. This function performs this type of averaging.
+      * The averaging is based on the choice in @p by_log. Averaging is conducted
+      * over the gamma functions of phases given in @gamma_inputs.second, with
+      * parameter_values stand for values of all the phases. Unlike the average_value
+      * function defined for compositions, averaging in this function is calculated base
+      * on gamma functions and the change of variables on the trajectory of phase boundaries.
+      * Thus on a single phase boundary, values of variables change stepwise from one phase
+      * to the other. And this is achieved by the definition of gamma function, which varies
+      * between 0 and 1.
+      */
+      double phase_average_value (const std::pair<std::vector<double>, const std::vector<unsigned int>> &gamma_inputs,
+                                  const std::vector<double> &parameter_values,
+                                  const unsigned int composition,
+                                  bool by_log=false);
```

Comment:
Same comment as above, I would suggest a different interface to make things easier to understand. What about the following:

```suggestion
      double phase_average_value (const std::vector<double> &parameter_values,
                                  const std::vector<double> &phase_function_values, 
                                  const std::vector<unsigned int> &n_phases_per_composition,
                                  const unsigned int composition_index,
                                  const MaterialAveraging::AveragingOperation operation = MaterialAveraging::arithmetic_average);
```

See the comment inside the function about how to use the `MaterialAveraging::AveragingOperation`.